org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/ASTConverter.java

```diff
@@ -2073,9 +2073,52 @@ public Expression convert(org.eclipse.jdt.internal.compiler.ast.Expression expre
 		if (expression instanceof org.eclipse.jdt.internal.compiler.ast.SwitchExpression) {
 			return convert((org.eclipse.jdt.internal.compiler.ast.SwitchExpression) expression);
 		}
+		if (expression instanceof org.eclipse.jdt.internal.compiler.ast.TemplateExpression) {
+			return convert((org.eclipse.jdt.internal.compiler.ast.TemplateExpression) expression);
+		}
+		if (expression instanceof org.eclipse.jdt.internal.compiler.ast.StringTemplate) {
+			return convert((org.eclipse.jdt.internal.compiler.ast.StringTemplate) expression);
+		}
 		return null;
 	}
+	public Expression convert(org.eclipse.jdt.internal.compiler.ast.TemplateExpression expression) {
+		StringTemplateExpression templateExpr = new StringTemplateExpression(this.ast);
+		if (this.resolveBindings) {
+			recordNodes(templateExpr, expression);
+		}
+		templateExpr.setProcessor(convert(expression.processor));
+		templateExpr.setTemplate(convert(expression.template));
+
+		int startPosition = expression.sourceStart;
+		int sourceEnd= expression.sourceEnd;
+		templateExpr.setSourceRange(startPosition, sourceEnd - startPosition + 1);
+		return templateExpr;
+	}
+	public StringTemplate convert(org.eclipse.jdt.internal.compiler.ast.StringTemplate template) {
+		StringTemplate strTemplate = new StringTemplate(this.ast);
+		if (this.resolveBindings) {
+			recordNodes(strTemplate, template);
+		}
+		List<Expression> fragments = strTemplate.fragments();
+		for (org.eclipse.jdt.internal.compiler.ast.StringLiteral fragment : template.fragments()) {
+			if (fragment instanceof org.eclipse.jdt.internal.compiler.ast.TextBlock) {
+				fragments.add(convert((org.eclipse.jdt.internal.compiler.ast.TextBlock) fragment));
+			} else {
+				fragments.add(convert(fragment));
+			}
+		}
+		List<Expression> expressions = strTemplate.expressions();
+		for (org.eclipse.jdt.internal.compiler.ast.Expression exp : template.values()) {
+			if (exp == null)
+				continue;
+			expressions.add(convert(exp));
```

Comment:
This check for null and continue could result in DOM expressions being not the right size. NullLiteral if constructed by Parser would work better here